### DISCUSSION TITLE
Is there a way to just populate an optional field with a None value if its validation fails?

### DISCUSSION THREAD

**Discussion by davidedigrande** [pydantic/pydantic#3748] [Question]
Hello.

Is there a way to just populate an optional field with a None value if its validation fails?

Seemed a trivial feature to me, but apparently I cannot figure out by myself hot to do it, and I could not find help in docs or other discussions online. So:

Let's say I have this model (code below) and the vat number validation fails. Pydantic as a default behaviour raises an exception and the Person object is not built. 

What I want to do is, considered it's an optional value,  I just want to have it as a None value instead if its validation fails and have the Person object built correctly.

I could accomplish this by adding a validator with a try-except and return None, but this doesn't look like an elegant solution to me.

How can accomplish this in a better way? Is there any model config I can use?

Thanks!

```{r, engine='python', count_lines}

class VatNumber(BaseModel):
    value: str

    @validator("value", pre=True)
    def validate_vat(cls, v):
        # do stuff
        return v

class Person(BaseModel):
    first_name: str = Field(..., alias="Name")
    second_name: str = Field(..., alias="Second Name")
    vat_number: Optional[VatNumber] = Field(alias="VAT")

```

**Comment by VianneyMI** [comment]
You can implement a validator like the one below :
```
@validator('value', pre=True)
def default_to_none(cls, v):
  
     if not isinstance(v, int):
         return None
     return v
```
